### PR TITLE
fix repeated primitive fields using different encodings when converting between OTLP bytes & OTAP

### PR DESCRIPTION
**Pull request by albertlockett**
part of #1399 

This fixes an issue caused by a mismatch encodings used repeated primitive proto fields. There are three fields of this type:
- HistogramDataPoint bucket_counts (repeated fixed64) & explicit_bounds (repeated double)
- ExpHistogramDataPoint Bucket bucket_counts (repeated uint64)

When we encoded OTAP -> OTLP bytes, we were writing the "expanded" encoding, but when decoding OTLP Bytes -> OTAP, we were expecting the "packed" encoding (for more details on the encodings see https://protobuf.dev/programming-guides/encoding/#repeated).

This corrects the encoding of OTAP -> OTLP bytes to use "packed", which is the default since proto Edition 2023 & in proto3 (it is also what prost uses).

In a followup PR, I will add correct the logic in our decoder to handle the expanded decoding and additional edge cases for packed encoding.

### CODE REVIEW

File: rust/otap-dataflow/crates/pdata/src/payload.rs [hunk 1]
```diff
@@ -415,59 +415,77 @@ mod test {
...
         };
 
         let result = ExportMetricsServiceRequest::decode(bytes.as_ref()).unwrap();
         assert_eq!(otlp_service_request, result);
+
+        // check that we can also re-decode the OTLP proto bytes that we encode directly
+        // from OTAP, that we get the same result
+        let pdata: OtapPayload = otlp_bytes.into();
+        let otap_batch2: OtapArrowRecords = pdata.try_into().unwrap();
+        assert_eq!(otap_batch, otap_batch2);
```
Comment (albertlockett) [inline rust/otap-dataflow/crates/pdata/src/payload.rs:488]: This test was failing before the code was fixed.

I added the same assertions to the other signal types as well for good measure.

**Review by jmacd — APPROVED** [review]
